### PR TITLE
Restore visual separator between mover buttons when show button label is on

### PR DESCRIPTION
**Pull request by afercia**
Follow-up to https://github.com/WordPress/gutenberg/pull/49556

Part of an effort to remove all the CSS order properties from the codebase.

## What?

While inspecting the CSS order properties, I noticed https://github.com/WordPress/gutenberg/pull/49556 removed the visual separator between the Move Up and Move Down block toolbar buttons when the 'Show button text labels' preference is enabled.

From an usability and accessibility perspective thatn's nod ideal as there is no visual separation between the buttons. It's not clear where the buttons boundary is and they just look like a single button with some multi line text within it, much like the other buttons in the toolbar.

Screenshot:

before

## Why?

There must be a clear separation between the two buttons.

## How?

- Restores the visual separator making sure it doesn't alter the toolbar height.
- Removes no longer used CSS `order` properties.
- Removes other CSS leftovers.

## Testing Instructions

- Go to the post editoror site editor > Options > Preferences > Accessibility tab
- Enable the 'Show button text labels' preference.
- Add a Group block with three paragrsphs inside of it.
- Select the second paragraph.
- Observe the mover buttons are visually separated bu the means of a horizontal line.
- Check the block toolbar height is 48 pixels, as expected.
- Check the parent selector button to select the Group block looks as expected (see glitches priginally fixed in https://github.com/WordPress/gutenberg/pull/49556)
- Enable the 'Top toolbar' from the Options menu.
- Observe the movers separator styls uses a lighter gray and some left / right spacing. See second screenshot.

## Screenshots or screencast 

Bock toolbar, zoomed in:

after

Top toolbar:

Screenshot 2024-01-08 at 12 41 04

### CODE REVIEW

**Review by t-hamano — COMMENTED** [review]
Thank you for the follow-up! I think this change makes sense, but since the `.block-editor-block-mover__move-button-container` selector already exists, how about changing it like this:

```diff
diff --git a/packages/block-editor/src/components/block-toolbar/style.scss b/packages/block-editor/src/components/block-toolbar/style.scss
index ae460faf5b..832988b416 100644
--- a/packages/block-editor/src/components/block-toolbar/style.scss
+++ b/packages/block-editor/src/components/block-toolbar/style.scss
@@ -203,6 +203,24 @@
 
        .block-editor-block-mover .block-editor-block-mover__move-button-container {
                width: auto;
+
+               @include break-small() {
+                       position: relative;
+
+                       &::before {
+                               content: "";
+                               display: block;
+                               height: $border-width;
+                               width: 100%;
+                               background: $gray-900;
+                               position: absolute;
+                               top: 50%;
+                               left: 50%;
+                               // With Top toolbar on, this separator has a smaller width. Translating both
+                               // X and Y axis allows to make the separator alwasy centered regardless of its width.
+                               transform: translate(-50%);
+                       }
+               }
        }
 
        .block-editor-block-mover.is-horizontal {
@@ -231,26 +249,6 @@
                padding-right: $grid-unit-15;
        }
 
-       @include break-small() {
-               .block-editor-block-mover__move-button-container {
-                       position: relative;
-
-                       &::before {
-                               content: "";
-                               display: block;
-                               height: $border-width;
-                               width: 100%;
-                               background: $gray-900;
-                               position: absolute;
-                               top: 50%;
-                               left: 50%;
-                               // With Top toolbar on, this separator has a smaller width. Translating both
-                               // X and Y axis allows to make the separator alwasy centered regardless of its width.
-                               transform: translate(-50%);
-                       }
-               }
-       }
-
        .block-editor-block-contextual-toolbar .block-editor-block-mover.is-horizontal .block-editor-block-mover-button.blo
ck-editor-block-mover-button {
                width: auto;
        }
```

**Comment by afercia** [comment]
@t-hamano makes totally sense. Thanks for your review 🙏

File: packages/edit-post/src/components/header/style.scss [hunk 1]
```diff
@@ -198,14 +198,24 @@
 	.edit-post-header__toolbar .block-editor-block-mover {
 		border-left: none;
 
+		// Modified group borders
```
Comment (t-hamano) [inline packages/edit-post/src/components/header/style.scss]: This comment doesn't seem to be related to the actual pseudo-element style, so how about deleting this comment?

File: packages/edit-site/src/components/header-edit-mode/style.scss [hunk 2]
```diff
@@ -145,14 +145,24 @@ $header-toolbar-min-width: 335px;
 	.block-editor-block-mover {
 		border-left: none;
 
+		// Modified group borders
```
Comment (t-hamano) [inline packages/edit-site/src/components/header-edit-mode/style.scss]: This comment doesn't seem to be related to the actual pseudo-element style, so how about deleting this comment?

File: packages/block-editor/src/components/block-toolbar/style.scss [hunk 3]
```diff
@@ -203,6 +203,24 @@
 
 	.block-editor-block-mover .block-editor-block-mover__move-button-container {
 		width: auto;
+
+		@include break-small() {
+			position: relative;
+
+			&::before {
+				content: "";
+				display: block;
```
Comment (t-hamano) [inline packages/block-editor/src/components/block-toolbar/style.scss]:
```suggestion
```

I think this property can be omitted if the element is absolutely positioned.

File: packages/block-editor/src/components/block-toolbar/style.scss [hunk 4]
```diff
@@ -203,6 +203,24 @@
...
+				height: $border-width;
+				width: 100%;
+				background: $gray-900;
+				position: absolute;
+				top: 50%;
+				left: 50%;
+				// With Top toolbar on, this separator has a smaller width. Translating both
+				// X and Y axis allows to make the separator alwasy centered regardless of its width.
+				transform: translate(-50%);
```
Comment (t-hamano) [inline packages/block-editor/src/components/block-toolbar/style.scss]:
```suggestion
				transform: translate(-50%, 0);
				margin-top: -$border-width * 0.5;
```

I think `translate(-50%);` means `translate(-50%, 0);`. Strictly speaking, it is not vertically centered.

I tried changing it to `translate(-50%, -50%)` to align it vertically, but the border seems to bleed.

image

Instead, I tried using a negative margin to shift the border up by half the height value. This approach draws a solid border in my environment. Could you test if this approach works in your environment?

Comment (afercia) [reply packages/block-editor/src/components/block-toolbar/style.scss]: > I think translate(-50%); means translate(-50%, 0);. Strictly speaking, it is not vertically centered.

Right.

> the border seems to bleed.

Are you on Windows or some GNU-Linux distro? What browser are you using? What display type and resolution?

> I tried using a negative margin to shift the border up by half the height value. This approach draws a solid border in my environment.

The root problem is that the toolbar height is 48px. The mover buttons height is 24px. A vertically centered 1 pixel horizontal line would need to be placed at exactly 23.5 pixels. However, a physical pixel can't be split in two. Some operating systems / browsers / displays (retina resolution, sub-pixel rendering) may render that more nicely than others. The 'bleeding' depends on the ability to position an element to 'half a pixel'. In environments where a CSS pixel is actually more than one physical pixel that may work better. In your screenshot, the horizontal line is actually 2 pixels, apparently that's because your environment isn't able to render sub-pixels.

The top margin approach actually computes to `margin-top: -0.5px;`. In your environment it may preserve the height of the line to 1 pixel but I guess the line isn't vertically centered.

Environments that support true subpixel rendering won't trigger this issue. The only reliable way to avoid a greyish blurred line or its rendered height to be altered would be to make sure it is moved to a 'full pixel' e.g. by making the toolbar height an odd number.

Maybe `translateZ(0)` may help prevent the blurriness because it promotes the element to its own layer but there's no guarantee the line will be vertically centered. Also `margin-top: -0.5px;` doesn't guarantee the line will be vertically centered but I'd think that's batter than a blurred line.

For the records, this is how the translateY approach looks in my environment (zoomed in): no blurring or doubled height:

Screenshot 2024-01-11 at 10 45 07

File: packages/edit-post/src/components/header/style.scss (hunk 1 above)
Comment (afercia) [reply packages/edit-post/src/components/header/style.scss]: This pseudo element is actually the modified border of the toolbar group when the 'Top toolbar' option is enabled:

Screenshot 2024-01-11 at 14 58 11

**Review by t-hamano — APPROVED** [review]
LGTM! 

> > the border seems to bleed.
> 
> Are you on Windows or some GNU-Linux distro? What browser are you using? What display type and resolution?

I am checking with Chrome browser on Windows OS. I have an external display connected to my laptop, and both have 1920 x 1080  resolution. I wasn't sure if these displays support sub-pixel rendering.

> The top margin approach actually computes to `margin-top: -0.5px;`. In your environment it may preserve the height of the line to 1 pixel but I guess the line isn't vertically centered.

So should we remove `margin-top: -0.5px` before merging this PR?

**Comment by afercia** [comment]
> So should we remove `margin-top: -0.5px` before merging this PR?

I think we should keep it. It works on environments that support true subpixel rendering.
